auth/auth/auth.py

```diff
@@ -806,7 +849,16 @@ async def on_startup(app):
     await db.async_init(maxsize=50)
     app['db'] = db
     app['client_session'] = httpx.client_session()
-    app['flow_client'] = get_flow_client('/auth-oauth2-client-secret/client_secret.json')
+
+    credentials_file = '/auth-oauth2-client-secret/client_secret.json'
+    if CLOUD == 'azure':
+        app['flow_client'] = AzureFlow(credentials_file)
+    else:
+        assert CLOUD == 'gcp'
+        app['flow_client'] = GoogleFlow(credentials_file)
+
+    with open('/auth-oauth2-client-secret/hailctl_client_secret.json', 'r', encoding='utf-8') as f:
+        app['hailctl_client_config'] = json.loads(f.read())
```

Comment:
Yes, but more broadly, the point of doing `hailctl auth login` is so that you can subsequently auth to hail (not the cloud in general). So what's the point of even doing that if you can't talk to hail.

There is a broader point here of whether we want to use a desktop login credentials for *all* of the cloud operations that the hail python library performs, instead of stealing gcloud application-default credentials. Doing that would motivate embedding a client secret into the package, so they can read from cloud storage without talking to the Hail Service. But that feels like a different discussion and I rather like sending it over the wire because then the `auth` service doesn't have to track the N client secrets that we have distributed with various versions.